docs/confluence/advanced/methods.md

```diff
@@ -1,11 +1,228 @@
+
+Internal Function Summaries
+===========================
+
+Summaries Always Inserted
+-------------------------
+
+Summaries for **external** functions are only inserted when an implementation
+of that function cannot be found, and so we default to some summary that we,
+the verifier, sees fit. _However_, internal functions can always be found and
+so it only makes sense to force-replace the body of the function (as opposed to
+filling in for one that could not be found).‌
+
+Feature Limitations:
+--------------------
+
+*   Function must have primitive parameter types and return types (`bool`,
+    `address`, `uintX`, `bytesX`)
+    
+*   Functions must be pure
+    
+
+Allowed Summaries
+-----------------
+
+Not all summaries make sense in the context of an internal function. Only the
+following summaries are allowed:
+
+*   `ALWAYS(X)` the summary always returns `X` and has no side-effects
+    
+*   `CONSTANT` the summary always returns the same constant and has no side
+    effects
+    
+*   `NONDET` the summary returns a havoc'd value
+    
+*   `Ghost` the summary returns the value return by the given ghost function
```

Comment:
CVL Function should be allowed too right?

Reply:
confluence docs are deprecated.